test/adjoint.jl

```diff
@@ -249,10 +249,17 @@ ū032,adj32 = adjoint_sensitivities_u0(sol,Tsit5(),dg,t,abstol=1e-14,
 
 ū04,adj4 = adjoint_sensitivities_u0(sol,Tsit5(),dg,t,abstol=1e-14,
                                     sensealg=InterpolatingAdjoint(checkpointing=true),
+                                    checkpoints=sol.t[1:10:end],
                                     reltol=1e-14,iabstol=1e-14,ireltol=1e-12)
 
+@test_throws Any adjoint_sensitivities_u0(sol,Tsit5(),dg,t,abstol=1e-14,
```

Comment:
why does this throw?

Reply:
I am not exactly sure, but coarsening checkpoints seems to help.

```julia
julia> adjoint_sensitivities_u0(sol,Tsit5(),dg,t,abstol=1e-14,
                                sensealg=InterpolatingAdjoint(checkpointing=true),
                                checkpoints=sol.t[1:5:end],
                                reltol=1e-14,iabstol=1e-14,ireltol=1e-12)
length(tstops) = 794
ERROR: A step rejection (around t=9.999833282928266) happened at a checkpoint, please coarsen the checkpoints to resolve this error.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] (::DiffEqSensitivity.ODEInterpolatingAdjointSensitivityFunction{Array{Float64,1},Array{Float64,1},Array{Float64,1},Nothing,DiffEqDiffTools.ParamJacobianWrapper{ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Array{Float64,1}},Nothing,Nothing,Nothing,Nothing,Array{Float64,2},Array{Float64,2},ForwardDiff.JacobianConfig{ForwardDiff.Tag{DiffEqDiffTools.ParamJacobianWrapper{ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Array{Float64,1}},Float64},Float64,4,Tuple{Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.ParamJacobianWrapper{ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Array{Float64,1}},Float64},Float64,4},1},Array{ForwardDiff.Dual{ForwardDiff.Tag{DiffEqDiffTools.ParamJacobianWrapper{ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Float64,Array{Float64,1}},Float64},Float64,4},1}}},ODESolution{Float64,2,Array{Array{Float64,1},1},Nothing,Nothing,Array{Float64,1},Array{Array{Array{Float64,1},1},1},ODEProblem{Array{Float64,1},Tuple{Float64,Float64},true,Array{Float64,1},ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{,Tuple{}}},DiffEqBase.StandardODEProblem},Tsit5,OrdinaryDiffEq.InterpolationData{ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Array{Array{Float64,1},1},Array{Float64,1},Array{Array{Array{Float64,1},1},1},OrdinaryDiffEq.Tsit5Cache{Array{Float64,1},Array{Float64,1},Array{Float64,1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}},DiffEqBase.DEStats},DiffEqSensitivity.CheckpointSolution{ODESolution{Float64,2,Array{Array{Float64,1},1},Nothing,Nothing,Array{Float64,1},Array{Array{Array{Float64,1},1},1},ODEProblem{Array{Float64,1},Tuple{Float64,Float64},true,Array{Float64,1},ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{,Tuple{}}},DiffEqBase.StandardODEProblem},Tsit5,OrdinaryDiffEq.InterpolationData{ODEFunction{true,typeof(fb),UniformScaling{Bool},Nothing,Nothing,typeof(jac),Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing,Nothing},Array{Array{Float64,1},1},Array{Float64,1},Array{Array{Array{Float64,1},1},1},OrdinaryDiffEq.Tsit5Cache{Array{Float64,1},Array{Float64,1},Array{Float64,1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}},DiffEqBase.DEStats},Array{Tuple{Float64,Float64},1}},Nothing,InterpolatingAdjoint{0,true,Val{:central}}})(::Array{Float64,1}, ::Array{Float64,1}, ::Array{Float64,1}, ::Float64) at /home/scheme/.julia/dev/DiffEqSensitivity/src/local_sensitivity/interpolating_adjoint.jl:114
...
```
```julia
julia> adjoint_sensitivities_u0(sol,Tsit5(),dg,t,abstol=1e-14,
                                sensealg=InterpolatingAdjoint(checkpointing=true),
                                checkpoints=sol.t[1:50:end],
                                reltol=1e-14,iabstol=1e-14,ireltol=1e-12)
length(tstops) = 80
([-87.94879817757926, -22.488414141679055], [25.500628335557487 -77.25509369834168 93.53213195229974 -286.99409835734303])
```

Reply:
floating point issues?

Reply:
```julia
    @show interval, t
    interval[1] <= t <= interval[2] || error("A step rejection (around t=$t) happened at a checkpoint, please coarsen the checkpoints to resolve this error.")
```
gives
```julia
julia> adjoint_sensitivities_u0(sol,Tsit5(),dg,t,abstol=1e-14,
                                sensealg=InterpolatingAdjoint(checkpointing=true),
                                checkpoints=sol.t[1:5:end],
                                reltol=1e-14,iabstol=1e-14,ireltol=1e-12)
(interval, t) = ((9.995909610141942, 10.0), 10.0)
(interval, t) = ((9.995909610141942, 10.0), 10.0)
(interval, t) = ((9.978207721677997, 9.995909610141942), 9.995588428138461)
(interval, t) = ((9.960645036521049, 9.978207721677997), 9.999833282928266)
ERROR: A step rejection (around t=9.999833282928266) happened at a checkpoint, please coarsen the checkpoints to resolve this error.
Stacktrace:
 [1] error(::String) at ./error.jl:33
...
```